src/parser/instruction.rs

```diff
@@ -814,6 +817,38 @@ mod tests {
         assert_eq!(parsed, expected);
     }
 
+    #[test]
+    fn parse_swap_phases() {
+        let input =
+            LocatedSpan::new(r#"SWAP-PHASES 0 "rf" 1 "rf"; SWAP-PHASES 1 2 3 "rf" 4 5 6 "rf""#);
+        let tokens = lex(input).unwrap();
+        let (remainder, parsed) = parse_instructions(&tokens).unwrap();
+        let expected = vec![
+            Instruction::SwapPhases(SwapPhases {
+                frame_1: FrameIdentifier {
+                    name: String::from("rf"),
+                    qubits: vec![Qubit::Fixed(0)],
+                },
+                frame_2: FrameIdentifier {
+                    name: String::from("rf"),
+                    qubits: vec![Qubit::Fixed(1)],
+                },
+            }),
+            Instruction::SwapPhases(SwapPhases {
+                frame_1: FrameIdentifier {
+                    name: String::from("rf"),
+                    qubits: vec![Qubit::Fixed(1), Qubit::Fixed(2), Qubit::Fixed(3)],
+                },
+                frame_2: FrameIdentifier {
+                    name: String::from("rf"),
+                    qubits: vec![Qubit::Fixed(4), Qubit::Fixed(5), Qubit::Fixed(6)],
+                },
+            }),
+        ];
+        assert_eq!(remainder.len(), 0);
```

Comment:
Minor suggestion:

```suggestion
        assert!(remainder.is_empty());
```

Reply:
Isn't it nice to know what the length is if it's not zero?